src/python/pants/engine/legacy/options_parsing.py

```diff
@@ -0,0 +1,60 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+from pants.build_graph.build_configuration import BuildConfiguration
+from pants.engine.rules import RootRule, rule
+from pants.engine.selectors import Select
+from pants.init.options_initializer import BuildConfigInitializer, OptionsInitializer
+from pants.option.options_bootstrapper import OptionsBootstrapper
+from pants.util.objects import datatype
+
+
+class OptionsParseRequest(datatype(['args', 'env'])):
+  """Represents a request for Options computation."""
+
+  @classmethod
+  def create(cls, args, env):
+    assert isinstance(args, (list, tuple))
+    return cls(
+      tuple(args),
+      tuple(env.items() if isinstance(env, dict) else env)
+    )
+
+
+class Options(datatype(['options', 'build_config'])):
+  """Represents the result of an Options computation."""
+
+
+@rule(OptionsBootstrapper, [Select(OptionsParseRequest)])
+def reify_options_bootstrapper(parse_request):
+  options_bootstrapper = OptionsBootstrapper(
+    env=dict(parse_request.env),
+    args=parse_request.args
+  )
+  # TODO: Once we have the ability to get FileContent for arbitrary
+  # paths outside of the buildroot, we can invert this to use
+  # OptionsBootstrapper.produce_and_set_bootstrap_options() which
+  # will yield lists of file paths for use as subject values and permit
+  # us to avoid the direct file I/O that this rule currently requires.
+  options_bootstrapper.construct_and_set_bootstrap_options()
```

Comment:
If this requires file IO, we should technically move the method call out, and then pass in the result as a field of `OptionsParseRequest`, to avoid doing IO inside a rule. As it stands, this will be memoized, and then options changes will be ignored.

Reply:
yeah, realized that and was just punting here with the TODO - but sure, makes sense.